www/mediadevices.js

```diff
@@ -20,58 +20,51 @@
 */
 /* globals Promise, cordova, MediaStream */
 var exec = cordova.require('cordova/exec');
+var channel = require('cordova/channel');
 var flagConstraints = true;
 var flagDevices = true;
-
 var mediaDevices = {
-    _devices: null
-};
-
-var supportedConstraints = {
-    width: true,
-    height: true,
-    aspectRatio: true,
-    frameRate: true,
-    facingMode: true,
-    volume: true,
-    sampleRate: true,
-    sampleSize: true,
-    echoCancellation: true,
-    latency: true,
-    channelCount: true,
-    deviceId: true,
-    groupId: true
+    _devices: [],
+    _supportedConstraints: {
+        width: true,
+        height: true,
+        aspectRatio: true,
+        frameRate: true,
+        facingMode: true,
+        volume: true,
+        sampleRate: true,
+        sampleSize: true,
+        echoCancellation: true,
+        latency: true,
+        channelCount: true,
+        deviceId: true,
+        groupId: true
+    }
 };
 
 mediaDevices.getSupportedConstraints = function () {
-    var success = function (constraints) {
-        console.log('constraints: ' + JSON.stringify(constraints));
-        supportedConstraints = constraints;
+    var successConstraints = function (constraints) {
+        mediaDevices._supportedConstraints = constraints;
+        flagConstraints = false;
     };
-
-    // assign new values returned from native ios ; until then default values returned
-    if (flagConstraints) {
-        exec(success, null, 'Stream', 'getSupportedConstraints', []);
+    if (!flagConstraints) {
+        return this._supportedConstraints;
+    } else {
+        exec(successConstraints, null, 'Stream', 'getSupportedConstraints', []);
```

Comment:
@maverickmishra they only thing that needs to be changes is after the exec call we'll need to:

```
return this._supportedConstraints;
```

This will return the default constraints as per the spec. Without a return statement here the function will return an undefined.

Reply:
Had missed it for `mediaDevices.enumerateDevices()` as well. Fixed it now